rust/vectordb/src/table.rs

```diff
@@ -187,7 +187,7 @@ impl Table {
     }
 
     /// Create index on the table.
-    pub async fn create_index(&mut self, index_builder: &impl VectorIndexBuilder) -> Result<()> {
+    pub async fn create_index(&self, index_builder: &impl VectorIndexBuilder) -> Result<Table> {
```

Comment:
If you make this take `&self` instead of `&mut self`, you are making the Rust and the Node API different. Node users get to write:

```ts
await table.delete("id = 42")
```

And their table is updated for them. Rust users would now have to remember to use the returned value:

```rust
let table = table.delete("id = 42").await?;
```

This seems unnecessary. I'd keep the Rust methods using `&mut` and change the `Arc<Table>` in Javascript to be `Table`. `Table` is cheap to clone as-is, so no need for it to be in an `Arc`.